### ISSUE TITLE
[TreeView] selected and focus indicators do not meet contrast ratio required by WCAG SC 1.4.11

### ISSUE THREAD

**Issue by amhci** [mui/mui-x#12559]
### Search keywords

contrast selected focus wcag

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: Go to https://mui.com/x/react-tree-view/ and **Edit in StackBlitz**. NOTE: I tried CodeSandbox too, but for this example at least, it gives an error FYI.

Steps:
1. Click on the first item "Data Grid". Note this node is now both selected and focused.
2. Press the down arrow key to move the focus to the second node. Note the first node is now selected and the second node is focused.

### Current behavior

The steps above demonstrate a node with both focus and selected state at the same time and then two individual nodes with the focused and selected state, respectively.

Both of these states use a semi-transparent background to indicate the state. The issue is that the rendered indicators for selected and focused states **_do not have enough contrast with their surroundings_**.

Refer to first paragraph under the ["User Interface Components" header](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast.html#user-interface-components) in the WCAG 2.2 Understanding Docs:
> Unless the control is inactive, any visual information provided that is necessary for a user to identify that a control is present and how to operate it must have a minimum 3:1 contrast ratio with the adjacent colors. Also, **_any visual information necessary to indicate state, such as whether a component is selected or focused must also ensure that the information used to identify the control in that state has a minimum 3:1 contrast ratio_**. 
(emphasis added)

Selected+focused vs. #FFF bg (1.27:1)
<img width="924" alt="Screenshot 2024-03-25 at 2 36 38 PM" src="https://github.com/mui/mui-x/assets/98350973/5b2e102b-446b-4227-b6b0-2529d149f969">

Focused vs. #FFF bg (1.28:1)
<img width="1034" alt="Screenshot 2024-03-25 at 2 36 06 PM" src="https://github.com/mui/mui-x/assets/98350973/622f2d74-553c-4b27-8a65-2891ea772bae">

Selected vs. #FFF bg (1.1:1)
<img width="805" alt="Screenshot 2024-03-25 at 2 35 10 PM" src="https://github.com/mui/mui-x/assets/98350973/2ce23bb5-84dd-46f7-ada7-454cf4b9b738">

These do not meet the 3:1 ratio required. Furthermore, using a change in background color for both states means you cannot easily distinguish between them or when both states are active on the same node. You would need to choose a color scheme that has a 3:1 contrast ratio between 3 separate colors and the background to achieve this.

### Expected behavior

I strongly encourage the team to adopt _different_ approaches to indicate selected and focused states. For example, I added a pseudo element that mimics the approach of indicating selection in the Tabs component via the existence of a thick line/rectangle. Then I used a complete inversion of the foreground and background to indicate focus. The result is you now meet contrast ratios for all states and you can now easily distinguish between them:
https://github.com/mui/mui-x/assets/98350973/f20db594-9773-4800-bee2-7dfa763f9872

Clearly my solution doesn't directly work for MUI's default styling, but you get the idea. I think the "thick line/rectangle" thing works well and you are _already_ doing this for Tabs. You could even re-use the same sliding animation (assuming that's sticking around for Material 3/MUI 6).

### Context

N/A

### Your environment

N/A -- all environments

### Order ID or Support key 💳 (optional)

77929

**Comment by michelengelen** [comment]
Hey @amhci,
thanks for raising this issue. I will add this to our board for the team to estimate.

**Comment by flaviendelangle** [comment]
@aarongarciah I think it's a global Material UI problem, not specific to the Tree View.
Do you had similar feedback in the past ? The Autocomplete for instance has similar contrasts on the hover and selected states.

cc @LukasTy @noraleonte

**Comment by michelengelen** [comment]
Grooming note: This is a global problem in material-ui in general. The workaround for now is to customize the colors as needed to accommodate for the correct ratio. @silviuaavram is currently working on a high contrast theme in material-ui
